src/js/messaging/utils/helpers.js

```diff
@@ -1,14 +1,18 @@
 import _ from 'lodash';
 import moment from 'moment';
 
-export function createQueryString(query) {
+export function createQueryString(query, camelCase = true) {
   const segments = [];
 
   for (const key of Object.keys(query)) {
     // Linter only accepts camelCase keys, but API only
     // recognizes snake_case for query string parameters.
-    const formattedKey = _.snakeCase(key);
-    segments.push(`${formattedKey}=${query[key]}`);
+    if (camelCase) {
```

Comment:
OTOH, it makes sense if it's saying that the query keys are typically camel cased, but the transformation is for anything you want to make snake case.